io/src/pcd_io.cpp

```diff
@@ -623,57 +593,27 @@ pcl::PCDReader::readBodyBinary (const unsigned char *map, pcl::PCLPointCloud2 &c
     {
       for (uindex_t c = 0; c < cloud.fields[d].count; ++c)
       {
+#define SET_CLOUD_DENSE(CASE_LABEL)                                                    \
+  case CASE_LABEL: {                                                                   \
+    if (!isValueFinite<pcl::traits::asType_t<CASE_LABEL>>(cloud, i, point_size, d, c)) \
```

Comment:
isValueFinite calls std::isFinite but that one doesn't accept bools: https://en.cppreference.com/w/cpp/numeric/math/isfinite - so it will fail?

Reply:
There's this in your link:
```
4) A set of overloads or a function template accepting the arg argument of any integral type. Equivalent to (2) (the argument is cast to double).
```
This leads to [integral type](https://en.cppreference.com/w/cpp/types/is_integral) which in effect makes it always integral. I'll see if I can modify the `isValueFinite` to skip checking finiteness for integral inputs.